File: 0238-product-of-array-except-self/0238-product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int res[] = new int[nums.length];
        int prefix = 1;
        int suffix = 1;
        for(int i=0;i<nums.length;i++){
            res[i] = prefix;
            prefix *= nums[i];
        }
        for(int i=nums.length-1;i>=0;i--){
            res[i] *= suffix;
            suffix *= nums[i];
        }
        return res;
    }
}

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution(object):
    def productExceptSelf(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        res = [1]*len(nums)
        prefix = 1
        suffix = 1
        for i in range(len(nums)):
            res[i] = prefix
            prefix *= nums[i]
        for i in range(len(nums)-1,-1,-1):
            res[i] *= suffix
            suffix *= nums[i]
        return res